msbuild/Xamarin.MacDev.Tasks/Tasks/Unzip.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Bundler;
+using Xamarin.Localization.MSBuild;
+using Xamarin.MacDev;
+using Xamarin.Messaging.Build.Client;
+
+#nullable enable
+
+namespace Xamarin.MacDev.Tasks {
+	/// <summary>
+	/// This task will extract the specified zip file into the specified extraction path.
+	/// This task works on Windows too, but if the task encounters a symlink while extracting, an error will be shown.
+	/// </summary>
+	public class Unzip : XamarinTask, ITaskCallback {
+		[Required]
+		public string ZipFilePath { get; set; } = string.Empty;
```

Comment:
I think this should by an ITaskItem, otherwise the zip file won't be copied to the Mac when the task needs to be executed remotely

Reply:
@emaf fixed!